Delivered-To: bclouse91@gmail.com
Received: by 2002:adf:dc0e:0:0:0:0:0 with SMTP id t14-v6csp947500wri;
        Fri, 17 Aug 2018 10:19:09 -0700 (PDT)
X-Google-Smtp-Source: AA+uWPx7EkZDPbnqFdCg+Ay8FGqpUXksN+fvnbErqwwJcf07eTFdvisODJTg7VzbinJTw8sSiiFV
X-Received: by 2002:a50:b37a:: with SMTP id r55-v6mr43176098edd.145.1534526349395;
        Fri, 17 Aug 2018 10:19:09 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1534526349; cv=none;
        d=google.com; s=arc-20160816;
        b=qQ0Z33HbtIvf6HlFtgqkyPvQVbX/fU2oilrkyLzHChY4dzj/RnOlhwPc/vljUa0ejf
         Y3C+0Ic5Be+uq8ZAUpXD7I5GPAl44A5zr7SSBkb2ODAEJSQUehAo3FOlKuBDNPU7az8D
         7ZZxP7SpK8dDKGFXovM8orz6w7mgLsKF9FeJwGHdzTRZGLo7aIaK3slKXKzdb6EWo4wM
         SnsI+cTQs1AeeefQRoNvQmCRbIvffHOQ9s3q7xiTkS3of0LTAeWmzUg72onob5T98+I5
         9kXjTra9qmnbsNJ8R6j6oTCWq6NNgY0DzS4JQGizStYf4hr4wmU2zET6eIhi+q8JxI2V
         cy9g==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=message-id:date:subject:cc:to:from:dkim-signature
         :arc-authentication-results;
        bh=ZRgTb+/ZmyzmDIjvD9kf0q9HFICQoozWilgRrDmlZ3Q=;
        b=U6yftPC1p9VYS6acuLKWjmV/vEETXeDez6dBNqCi2IyynIa7QgrA6vxSXv4Pi8gZqK
         St/W/ouc3TEtEXnh4LvImB+zc5B4GD8C7mG7emn/FfU/yvGkLEQ7QL/bCBfiYRdK8IdA
         wQcng1IrogbFQLhZ+8th4B8HPj0zer2rDNCcrknlER6viqF+FWFUazBoM91GCEETHFzJ
         WCoT/S+2WDCCCXIGeCEF21MN1E394aGRHc+JT5fIHjmfg4MYaAD33bUnxTCW/EGMuxgg
         Qt728NSUWW6EBCwc8GjXjuQhUgjl8LXm0alSYcVC4hQXyzBFc6czfCDCE36ySholUaY9
         +scA==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass header.i=@agner.ch header.s=dkim header.b=s4OFnZ6o;
       spf=pass (google.com: domain of stefan@agner.ch designates 2a02:418:6a02::a2 as permitted sender) smtp.mailfrom=stefan@agner.ch
Return-Path: <stefan@agner.ch>
Received: from mail.kmu-office.ch (mail.kmu-office.ch. [2a02:418:6a02::a2])
        by mx.google.com with ESMTPS id d25-v6si3896954edb.351.2018.08.17.10.19.09
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Fri, 17 Aug 2018 10:19:09 -0700 (PDT)
Received-SPF: pass (google.com: domain of stefan@agner.ch designates 2a02:418:6a02::a2 as permitted sender) client-ip=2a02:418:6a02::a2;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@agner.ch header.s=dkim header.b=s4OFnZ6o;
       spf=pass (google.com: domain of stefan@agner.ch designates 2a02:418:6a02::a2 as permitted sender) smtp.mailfrom=stefan@agner.ch
Received: from trochilidae.toradex.int (unknown [46.140.72.82])
	by mail.kmu-office.ch (Postfix) with ESMTPSA id C743A5C08B6;
	Fri, 17 Aug 2018 19:19:08 +0200 (CEST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=agner.ch; s=dkim;
	t=1534526348;
	h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
	 to:to:cc:cc:mime-version:content-type:content-transfer-encoding:
	 in-reply-to:references; bh=ZRgTb+/ZmyzmDIjvD9kf0q9HFICQoozWilgRrDmlZ3Q=;
	b=s4OFnZ6omW22eLf01r0eqyj9AqMFVzPdttGx1dqhXquh5sGHPy7TdJCBcfxe4UFFtrjZuS
	74xFnUkcgjiwBoYNd3Xz3TDUzZqjnyN9l9ihQsh2QwYOakqe5SEPMXUGPItX6TY9P5W6c2
	gPUvlHM05wKBrQwbFpUIht0V9it6AZM=
From: Stefan Agner <stefan@agner.ch>
To: rmk@armlinux.org.uk
Cc: etnaviv@lists.freedesktop.org,
	christian.gmeiner@gmail.com,
	bclouse91@gmail.com,
	Stefan Agner <stefan@agner.ch>
Subject: [xf86-video-armada][PATCH v2 1/2] common: Make 24bbp support optional
Date: Fri, 17 Aug 2018 19:19:04 +0200
Message-Id: <20180817171905.14594-1-stefan@agner.ch>
X-Mailer: git-send-email 2.18.0

24bbp support has been removed in X.Org Server 1.20. Make
its support optional so this driver can be built for newer
X.Org Server versions.

Suggested-by: Christian Gmeiner <christian.gmeiner@gmail.com>
Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 common/unaccel.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/unaccel.c b/common/unaccel.c
index 15cc7d5..b0d807e 100644
--- a/common/unaccel.c
+++ b/common/unaccel.c
@@ -192,6 +192,7 @@ static void unaccel_fixup_tile(DrawablePtr pDraw, PixmapPtr *ppPix)
 {
 	PixmapPtr pNew, pPixmap = *ppPix;
 
+#ifdef FB_24_32BIT
 	if (pPixmap->drawable.bitsPerPixel != pDraw->bitsPerPixel) {
 		prepare_cpu_drawable(&pPixmap->drawable, CPU_ACCESS_RO);
 		pNew = fb24_32ReformatTile(pPixmap, pDraw->bitsPerPixel);
@@ -200,6 +201,7 @@ static void unaccel_fixup_tile(DrawablePtr pDraw, PixmapPtr *ppPix)
 		pDraw->pScreen->DestroyPixmap(pPixmap);
 		*ppPix = pPixmap = pNew;
 	}
+#endif
 
 	if (FbEvenTile(pPixmap->drawable.width * pPixmap->drawable.bitsPerPixel)) {
 		prepare_cpu_drawable(&pPixmap->drawable, CPU_ACCESS_RW);
-- 
2.18.0

